src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+import {CheckpointsUpgradeable} from "@openzeppelin-upgrades/contracts/utils/CheckpointsUpgradeable.sol";
+import {PausableUpgradeable} from "@openzeppelin-upgrades/contracts/security/PausableUpgradeable.sol";
+import {SignatureCheckerUpgradeable} from "@openzeppelin-upgrades/contracts/utils/cryptography/SignatureCheckerUpgradeable.sol";
+import {IERC1271Upgradeable} from "@openzeppelin-upgrades/contracts/interfaces/IERC1271Upgradeable.sol";
+
+/// @title ECDSA Stake Registry
+/// @notice THIS CONTRACT IS NOT AUDITED
+/// @notice Manages operator registration and quorum updates for an AVS using ECDSA signatures.
+/// @dev Extends OpenZeppelin's upgradeable Ownable and Pausable patterns to ensure upgradability and pause functionality.
+contract ECDSAStakeRegistry is
+    IERC1271Upgradeable,
+    OwnableUpgradeable,
+    PausableUpgradeable,
+    ECDSAStakeRegistryStorage
+{
+    using SignatureCheckerUpgradeable for address;
+    using CheckpointsUpgradeable for CheckpointsUpgradeable.History;
+
+    /// @dev Constructor to create ECDSAStakeRegistry.
+    /// @param _delegationManager Address of the DelegationManager contract that this registry interacts with.
+    constructor(
+        IDelegationManager _delegationManager
+    ) ECDSAStakeRegistryStorage(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    /// @notice Initializes the contract with the given parameters.
+    /// @param _serviceManager The address of the service manager.
+    /// @param _thresholdWeightBps The threshold weight in basis points.
+    /// @param _quorum The quorum struct containing the details of the quorum thresholds.
+    function initialize(
+        address _serviceManager,
+        uint256 _thresholdWeightBps,
+        Quorum memory _quorum
+    ) external initializer {
+        __ECDSAStakeRegistry_init(_serviceManager, _thresholdWeightBps, _quorum);
+    }
+
+    /// @notice Registers a new operator using a provided signature
+    /// @param _operatorSignature Contains the operator's signature, salt, and expiry
+    function registerOperatorWithSignature(
+        address _operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory _operatorSignature
+    ) external {
+        _registerOperatorWithSig(_operator, _operatorSignature);
+    }
+
+    /// @notice Deregisters an existing operator
+    function deregisterOperator() external {
+        _deregisterOperator(msg.sender);
+    }
+
+    /**
+     * @notice Updates the StakeRegistry's view of one or more operators' stakes adding a new entry in their history of stake checkpoints, 
+     * @dev Queries stakes from the Eigenlayer core DelegationManager contract
+     * @param _operators A list of operator addresses to update
+     */
+    function updateOperators(address[] memory _operators) external {
+        for (uint256 i; i < _operators.length; i++) {
+            _updateOperatorWeight(_operators[i]);
+        }
+    }
```

Comment:
seems like there needs to be a check for the operators being registered. perhaps safest to check in the internal function?

Reply:
yeah nice catch

Reply:
added a regression case for this

Reply:
I realized this was already handled in the `_updateOperatorWeight` function where operators that aren't registered either because they were deregistered immediately before the `_updateOperatorWeight` call or because they weren't registered in the first place can only push a weight of 0 to their history.

https://github.com/Layr-Labs/eigenlayer-middleware/blob/948538b93971c8384500e35a5ad07dfe0d2ccd88/src/unaudited/ECDSAStakeRegistry.sol#L230

The clarity of this code path could be improved, though